server/src/main/java/io/seata/server/session/FileBasedSessionManager.java

```diff
@@ -194,9 +197,30 @@ private void restore(List<TransactionWriteStore> stores, Map<Long, BranchSession
 
             }
         }
+        setMaxId(maxRecoverId);
 
     }
 
+    private long getMaxId(long maxRecoverId, SessionStorable sessionStorable) {
+        long currentId = 0;
+        if (sessionStorable instanceof GlobalSession) {
+            currentId = ((GlobalSession)sessionStorable).getTransactionId();
+        } else if (sessionStorable instanceof BranchSession) {
+            currentId = ((BranchSession)sessionStorable).getBranchId();
+        }
+
+        return maxRecoverId > currentId ? maxRecoverId : currentId;
+    }
+
+    private void setMaxId(long maxRecoverId) {
```

Comment:
getMaxId() contains ` maxRecoverId > currentId ? maxRecoverId : currentId` and setMaxId is single thread
maybe setMaxId should direct  set not use while or if

Reply:
fixed